### PR TITLE
RC2 ECB/CBC and PKCS#12 Integration

### PR DESCRIPTION
**Pull request by cconlon**
This PR adds RC2-ECB/CBC support and integration into PKCS#12 parsing functionality.

By default, OpenSSL generates .p12 bundles using pbeWithSHA1And40BitRC2-CBC encryption.  Prior to this PR, wolfSSL could not parse and decode .p12 bundles generated by the default openssl pkcs12 command.

This PR includes unit tests, wolfCrypt test code, and has been tested against the PKCS#12 example in wolfssl-examples/crypto/pkcs12 using the <wolfssl_root>/certs/test-servercert-rc2.p12 bundle.

### CODE REVIEW

File: wolfssl/wolfcrypt/rc2.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    RC2_BLOCK_SIZE   = 8
+};
+
+/* RC2 encryption and decryption */
+typedef struct Rc2 {
+    word32 keylen;      /* key length, octets */
+    word32 bits;        /* effective key length, bits */
+    ALIGN16 word16 key[RC2_MAX_KEY_SIZE/2];
+    ALIGN16 word32 reg[RC2_BLOCK_SIZE / sizeof(word32)];    /* for CBC mode */
+    ALIGN16 word32 tmp[RC2_BLOCK_SIZE / sizeof(word32)];    /* same         */
```
Comment (toddouska) [inline wolfssl/wolfcrypt/rc2.h:44]: For optimal structure packing / alignment the ALIGN16s should go at the top of the struct, biggest first.

File: wolfcrypt/src/rc2.c [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+        return BAD_FUNC_ARG;
+    }
+
+    if (sz == 0) {
+        return 0;
+    }
+
+    while (blocks--) {
+        XMEMCPY(rc2->tmp, in, RC2_BLOCK_SIZE);
+        wc_Rc2EcbDecrypt(rc2, out, (byte*)rc2->tmp, RC2_BLOCK_SIZE);
```
Comment (toddouska) [inline wolfcrypt/src/rc2.c:321]: Return code check?

File: wolfcrypt/src/rc2.c [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        return BAD_FUNC_ARG;
+    }
+
+    if (sz == 0) {
+        return 0;
+    }
+
+    while (blocks--) {
+        xorbuf((byte*)rc2->reg, in, RC2_BLOCK_SIZE);
+        wc_Rc2EcbEncrypt(rc2, (byte*)rc2->reg, (byte*)rc2->reg, RC2_BLOCK_SIZE);
```
Comment (toddouska) [inline wolfcrypt/src/rc2.c:297]: Return code check

File: wolfssl/wolfcrypt/rc2.h (hunk 1 above)
Comment (cconlon) [reply wolfssl/wolfcrypt/rc2.h:42]: Rearranged, thanks

File: wolfcrypt/src/rc2.c (hunk 2 above)
Comment (cconlon) [reply wolfcrypt/src/rc2.c]: Done, thanks.  Had been following the precedence of other CBC mode ops.

File: wolfcrypt/src/rc2.c (hunk 3 above)
Comment (cconlon) [reply wolfcrypt/src/rc2.c]: Done, thanks.  Had been following the precedence of other CBC mode ops.

**Review by toddouska — APPROVED** [review]

**Pull request merged**